Sources/NostrSDK/DirectMessageEncrypting.swift

```diff
@@ -0,0 +1,170 @@
+        }
+
+        let sections = Array(message.split(separator: "?"))
+
+        if sections.count != 2 {
+            throw DirectMessageEncryptingError.decryptionError
+        }
+
+        guard let encryptedContent = sections.first,
+              let encryptedContentData = Data(base64Encoded: String(encryptedContent)) else {
+            throw DirectMessageEncryptingError.decryptionError
+        }
+
+        guard let ivContent = sections.last else {
+            throw DirectMessageEncryptingError.decryptionError
+        }
+
+        let ivContentTrimmed = ivContent.dropFirst(3)
+
+        guard let ivContentData = Data(base64Encoded: String(ivContentTrimmed)) else {
+            throw DirectMessageEncryptingError.decryptionError
+        }
+        
+        guard let decryptedContentData = AESDecrypt(data: encryptedContentData.bytes, iv: ivContentData.bytes, sharedSecret: sharedSecret) else {
+            throw DirectMessageEncryptingError.decryptionError
+        }
+
+        guard let decryptedMessage = String(data: decryptedContentData, encoding: .utf8) else {
+            throw DirectMessageEncryptingError.decryptionError
+        }
+
+        return decryptedMessage
+    }
+
+    private func getSharedSecret(privateKey: PrivateKey, recipient pubkey: PublicKey) throws -> [UInt8] {
+        let privateKeyBytes = privateKey.dataRepresentation.bytes
+        let publicKeyBytes = preparePublicKeyBytes(from: pubkey)
+
+        let recipientPublicKey = try parsePublicKey(from: publicKeyBytes)
+        return try computeSharedSecret(using: recipientPublicKey, and: privateKeyBytes)
+    }
+
+    private func preparePublicKeyBytes(from pubkey: PublicKey) -> [UInt8] {
+        var bytes = pubkey.dataRepresentation.bytes
+        bytes.insert(2, at: 0)
+        return bytes
+    }
+
+    private func parsePublicKey(from bytes: [UInt8]) throws -> secp256k1_pubkey {
+        var recipientPublicKey = secp256k1_pubkey()
+        guard secp256k1_ec_pubkey_parse(secp256k1.Context.rawRepresentation, &recipientPublicKey, bytes, bytes.count) != 0 else {
+            throw DirectMessageEncryptingError.pubkeyParsing
+        }
+        return recipientPublicKey
+    }
+
+    private func computeSharedSecret(using publicKey: secp256k1_pubkey, and privateKeyBytes: [UInt8]) throws -> [UInt8] {
+        var sharedSecret = [UInt8](repeating: 0, count: 32)
+        var mutablePublicKey = publicKey
+        guard secp256k1_ecdh(secp256k1.Context.rawRepresentation, &sharedSecret, &mutablePublicKey, privateKeyBytes, { (output, x32, _, _) in
+            memcpy(output, x32, 32)
+            return 1
+        }, nil) != 0 else {
+            throw DirectMessageEncryptingError.sharedSecretCreation
+        }
+        return sharedSecret
+    }
+
+    private func AESDecrypt(data: [UInt8], iv: [UInt8], sharedSecret: [UInt8]) -> Data? {
+        return AESOperation(operation: CCOperation(kCCDecrypt), data: data, iv: iv, sharedSecret: sharedSecret)
```

Comment:
You don't need the word `return` if it's only one line inside the function. Same for the one below.